cabal-testsuite/PackageTests/Regression/T4154/install-time-with-constraint.test.hs

```diff
@@ -9,7 +9,9 @@ main = cabalTest $ withRepo "repo" $ do
 
   -- Temporarily disabled recording here because output is not stable
   recordMode DoNotRecord $ do
-      -- Constraining all uses of 'time' results in a cyclic dependency
-      -- between 'Cabal' and the new 'time'.
-      r <- fails $ cabal' "new-build" ["time", "--constraint=any.time==99999", "--dry-run"]
-      assertOutputContains "cyclic dependencies; conflict set: time:setup.Cabal, time:setup.time" r
+      r <- fails $ cabal' "new-build" ["time", "--constraint=any.time==99999", "--constraint=setup.Cabal installed", "--dry-run"]
```

Comment:
Why does this command need the `setup.Cabal installed` constraint now?  It seems like cabal can only choose the installed Cabal library now that it enforces a max setup Cabal version.  I thought that the two new-build commands should only differ by the constraint scope, to show that the constraint on `time` succeeds, but the constraint on `any.time` fails.

Reply:
@grayjay 

I was mostly paranoid and wanted the resulting output to be as deterministic as possible but I realize this wasn't probably the most principled way to retain the spirit of this test... 

...and I gladly take your offer on changing the tests after the PR is merged, as you clearly have a better understanding on how to write this regression test in an effective way. :-)

Reply:
I made a PR: #5186